accounting_reporting_core/src/main/java/org/cardanofoundation/lob/app/accounting_reporting_core/job/TxStatusUpdaterJob.java

```diff
@@ -0,0 +1,107 @@
+package org.cardanofoundation.lob.app.accounting_reporting_core.job;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import io.vavr.control.Either;
+import org.zalando.problem.Problem;
+
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.core.LedgerDispatchStatus;
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.core.TxStatusUpdate;
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.entity.TransactionEntity;
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.entity.report.ReportEntity;
+import org.cardanofoundation.lob.app.accounting_reporting_core.repository.ReportRepository;
+import org.cardanofoundation.lob.app.accounting_reporting_core.service.internal.LedgerService;
+import org.cardanofoundation.lob.app.accounting_reporting_core.service.internal.ReportService;
+import org.cardanofoundation.lob.app.accounting_reporting_core.service.internal.TransactionBatchService;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+@ConditionalOnProperty(value = "lob.accounting_reporting_core.enabled", havingValue = "true", matchIfMissing = true)
+public class TxStatusUpdaterJob {
+
+    private final Map<String, TxStatusUpdate> txStatusUpdatesMap = new ConcurrentHashMap<>();
+    private final LedgerService ledgerService;
+    private final TransactionBatchService transactionBatchService;
+    private final ReportRepository reportRepository;
+    private final ReportService reportService;
+
+    @Value("${lob.blockchain.tx-status-updater.max-map-size:1000}")
+    private int maxMapSize;
+
+    // This Job collects all TxStatusUpdate events and updates the transactions in the database
+    @Scheduled(
+            fixedDelayString = "${ob.blockchain.tx-status-updater.fixed_delay:PT30S}",
+            initialDelayString = "${lob.blockchain.tx-status-updater.delay:PT30S}")
+    @Transactional
+    public void execute() {
+        Map<String, TxStatusUpdate> updates;
+        synchronized (txStatusUpdatesMap) {
```

Comment:
in only very very special cases you need synchronized block for concurrent hash map, typically you don't